data/basics/overflow.go

```diff
@@ -173,28 +131,28 @@ func MulAIntSaturate(a MicroAlgos, b int) MicroAlgos {
 }
 
 // AddA adds 2 MicroAlgos values with overflow tracking
-func (t *OverflowTracker) AddA(a MicroAlgos, b MicroAlgos) MicroAlgos {
-	return MicroAlgos{Raw: t.Add(uint64(a.Raw), uint64(b.Raw))}
+func (t *OverflowTracker[T]) AddA(a, b MicroAlgos) MicroAlgos {
+	return MicroAlgos{Raw: uint64(t.Add(T(a.Raw), T(b.Raw)))}
```

Comment:
I think this is another reason not to make generic OverflowTracker. Imagine you created an `OverflowTracker[uint32]`, and used it to add to MicroAlgos values that are each "2^32+5". Both are cast to T *first*, so they become 5. Those get added by the OverflowTracker and do *not* overflow. Then the result is cast to u64 and returned.

Reply:
yea, the more thinking about `T` being tied to the `OverflowTracker`, the more it feels a bit off. The logic seems a bit distorted. I am reverting the change now.